src/Reindex.php

```diff
@@ -73,9 +73,16 @@ public function run(): Response
         return $this->_lastResponse;
     }
 
+    /**
+     * @param bool|string $value
+     *
+     * @phpstan-param bool|self::WAIT_FOR_COMPLETION_* $value
+     */
     public function setWaitForCompletion($value): void
     {
-        \is_bool($value) && $value = $value ? 'true' : 'false';
+        if (\is_bool($value)) {
+            $value = $value ? 'true' : 'false';
```

Comment:
From the ES documentation, this parameter is a boolean, no need  to use strings as parameters.
(while we still need to use "true" or "false" as strings in `->setParam()`
Would you adjust it? in that case the  `self::WAIT_FOR_COMPLETION_FALSE` can be removed (if not used)

Reply:
Should we trigger an user warning in case non-boolean are passed here?
Like an user_error or something like that?